app/models/entities/opportunity.rb

```diff
@@ -0,0 +1,60 @@
+class Entities::Opportunity < Maestrano::Connector::Rails::Entity
+  def self.connec_entity_name
+    'Opportunity'
+  end
+
+  def self.external_entity_name
+    'Deal'
+  end
+
+  def self.mapper_class
+    OpportunityMapper
+  end
+
+  def self.object_name_from_connec_entity_hash(entity)
+    entity['name']
+  end
+
+  def self.object_name_from_external_entity_hash(entity)
+    entity['name']
+  end
+
+  def self.references
+    %w(lead_id)
+  end
+
+  def before_sync(last_synchronization_date = nil)
+     @stages = @external_client.get_entities('stages')
+     @stages ||= []
+  end
+
+  def map_to_connec(entity)
+    mapped_entity = super
+    stage = @stages.find { |stage| stage['id'] == entity['stage_id'] }
+    mapped_entity[:sales_stage] = stage['name'] ? stage['name'] : "stage not found"
+    mapped_entity[:probability] = stage['likelihood'] ? stage['likelihood'] : 0
+    mapped_entity
+  end
+
+  def map_to_external(entity)
+     mapped_entity = super
+     stage = @stages.find { |stage| stage['name'].downcase == entity['sales_stage'].downcase}
+     stage ||= @stages.find { |stage| stage['likelyhood'] == entity['probability']}
```

Comment:
You could be even smarter and take the closest likelyhood (to be able to map as many possible input as possible)

For inspiration:
```ruby
xs = ["77.5", "80", "82.5", "85", "87.5", "90", "95", "100", "105"]
xs.min_by { |x| (x.to_f - 82.4).abs } 
#=> "82.5"
```